test/server/lds_api_test.cc

```diff
@@ -180,9 +190,41 @@ TEST_F(LdsApiTest, MisconfiguredListenerNameIsPresentInException) {
 
   EXPECT_CALL(listener_manager_, addOrUpdateListener(_, _, true))
       .WillOnce(Throw(EnvoyException("something is wrong")));
+  EXPECT_CALL(init_.initialized_, ready());
+
+  EXPECT_THROW_WITH_MESSAGE(
+      lds_->onConfigUpdate(listeners, ""), EnvoyException,
+      "Error adding/updating listener(s) invalid-listener: something is wrong");
+  EXPECT_CALL(request_, cancel());
+}
+
+TEST_F(LdsApiTest, ListenerCreationContinuesEvenAfterException) {
+  InSequence s;
+
+  setup();
+
+  Protobuf::RepeatedPtrField<envoy::api::v2::Listener> listeners;
+  std::vector<std::reference_wrapper<Network::ListenerConfig>> existing_listeners;
+
+  // Add 4 listeners - 2 valid and 2 invalid.
+  addListener(listeners, "valid-listener-1");
+  addListener(listeners, "invalid-listener-1");
+  addListener(listeners, "valid-listener-2");
+  addListener(listeners, "invalid-listener-2");
+
+  EXPECT_CALL(listener_manager_, listeners()).WillOnce(Return(existing_listeners));
+
+  EXPECT_CALL(listener_manager_, addOrUpdateListener(_, _, true))
+      .WillOnce(Return(true))
+      .WillOnce(Throw(EnvoyException("something is wrong")))
+      .WillOnce(Return(true))
+      .WillOnce(Throw(EnvoyException("something else is wrong")));
+
+  EXPECT_CALL(init_.initialized_, ready());
 
   EXPECT_THROW_WITH_MESSAGE(lds_->onConfigUpdate(listeners, ""), EnvoyException,
-                            "Error adding/updating listener invalid-listener: something is wrong");
+                            "Error adding/updating listener(s) invalid-listener-1: something is "
+                            "wrong, invalid-listener-2: something else is wrong");
```

Comment:
Can we have similar behavioral coverage to https://github.com/envoyproxy/envoy/pull/5806/files?

Reply:
Sorry. I did n't get it. which specific test case you are talking about ?

Reply:
I could only one additional empty update test - that I have added now if that  is what you are referring to. Otherwise LMK, will add.